### PR TITLE
re organize revamp

### PR DESCRIPTION
**Pull request by ilanDoron**

### CODE REVIEW

File: contractsSol5/KyberNetwork.sol [hunk 1]
```diff
@@ -478,297 +470,225 @@ contract KyberNetwork is Withdrawable3, Utils4, IKyberNetwork, ReentrancyGuard {
...
+            return (0, 0);
         }
 
-        tData.networkFeeWei = tData.tradeWei * tData.networkFeeBps / BPS * tData.feePayingReservesBps / BPS;
+        // trade wei and fees.
+        tData.networkFeeWei = tData.tradeWei * tData.networkFeeBps / BPS * tData.feeAccountedBps / BPS;
         tData.platformFeeWei = tData.tradeWei * tData.input.platformFeeBps / BPS;
 
-        require(tData.tradeWei >= (tData.networkFeeWei + tData.platformFeeWei), "fees exceed trade amt");
+        require(tData.tradeWei >= (tData.networkFeeWei + tData.platformFeeWei), "fees exceed trade anount");
```
Comment (ayobuenavista) [inline contractsSol5/KyberNetwork.sol:489]: Minor typo: `anount`

File: contractsSol5/KyberNetwork.sol [hunk 2]
```diff
@@ -478,297 +470,225 @@ contract KyberNetwork is Withdrawable3, Utils4, IKyberNetwork, ReentrancyGuard {
...
+        tradingReserves.rates = new uint[](numReserves);
+        tradingReserves.feeAccountedBpsDest = new uint[](numReserves);
+        tradingReserves.addresses = new IKyberReserve[](numReserves);
+
+        // iterate reserve list. validate data. calculate srcAmount according to splits and fee data.
+        for (uint i = 0; i < numReserves; i++) {
+            require(tradingReserves.splitsBps[i] > 0 && tradingReserves.splitsBps[i] <= BPS, "invalid split bps");
+            tradingReserves.addresses[i] = IKyberReserve(convertReserveIdToAddress(tradingReserves.ids[i]));
+            
+            tradingReserves.srcAmounts[i] = srcAmount * tradingReserves.splitsBps[i] / BPS;
```
Comment (manhlx3006) [inline contractsSol5/KyberNetwork.sol:588]: This part of src amounts is not correct for splitting trade (e2t) compare to our current logic.
For t2e, it is correct that `srcAmounts[i] = srcAmount * splitsBps[i] / BPS`
For e2t, if it is splitting trade, we compute total `feePayingReserveBps` (combine of t2e and e2t), then compute total `networkFeeWei`. After that we have `actualSrcAmount` for e2t trade (`tradeWei - networkFeeWei - platformFeeWei`, and `srcAmounts[i] = actualSrcAmount * splitsBps[i] / BPS`.


Take an example:
assume: networkFee: 0.25%, platformFee = 0
t2e: done, tradeWei = 1 eth, networkFeeWei = 0.25% * 1 = 0.0025 eth (feePayingReserveBps = 100%)
e2t: split into 2 reserves, each 50%, one reserve is paying fee, one isn't.

Current logic:
- feePayingReserveBps = 100% (t2e) + 50% (e2t) = 150%
- networkFeeWei = 1 * 0.25% * 150% = 0.00375 eth
- actualSrcAmount = 1 - 0.00375 = 0.99625 eth
- src amount for each reserve in e2t trade: 0.99625 * 50% = 0.498125 eth

New logic:
- srcAmount for e2t: 1 - 0.25% * 1 = 0.9975 eth
- srcAmount each reserve: srcAmounts[i] = srcAmount * 50% = 0.49875 eth
- srcAmount reserve not paying fee = 0.49875 eth
- srcAmount reserve paying fee = 0.49875 * (100% - 0.25%) = 0.497503125 eth (in line 593)

Comment (ilanDoron) [reply contractsSol5/KyberNetwork.sol:588]: OK

Comment (ilanDoron) [reply contractsSol5/KyberNetwork.sol:581]: fixed
with a bit new logic
See here
https://github.com/KyberNetwork/smart-contracts/pull/807/commits/4e5194171f76759b69b7b4c19731d88ec28851aa

last section

**Pull request merged**